File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/1/2025, 10:43:12 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
       # m=[]
        # for i in range(len(nums)):
        #     for j in range(len(nums)):
        #         if nums[i]+nums[j]==target:
        #           x.append(i)
        #           x.append(j)  
        #           return x
        # s=[[x,y] for x in range(len(nums)) for y in range(1,len(nums)) if nums[x]+nums[y]==target]

        # return s[0]
        
        # Create a dictionary to store the indices of elements
        num_indices = {}
        
        # Iterate through the list of numbers
        for i, num in enumerate(nums):
            # Calculate the complement
            complement = target - num
            
            # Check if the complement exists in the dictionary
            if complement in num_indices:
                # Return the indices of the current number and its complement
                return [num_indices[complement], i]
            
            # Store the index of the current number in the dictionary
            num_indices[num] = i
        
        # If no such pair is found, return an empty list
        return []

File: 1-TwoSum/1-TwoSum.java
// Last updated: 8/1/2025, 10:43:11 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] nums2 = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            nums2[i] = nums[i];
        }
        
        Arrays.sort(nums2);
        int i = 0;
        int j = nums.length - 1;
        int[] arr = new int[2];
        int n1 = -1;
        int n2 = -1;
        
        while (i < j) {
            if (nums2[i] + nums2[j] == target) {
                n1 = nums2[i];
                n2 = nums2[j];
                break;
            } else if (nums2[i] + nums2[j] < target) {
                i++;
            } else {
                j--;
            }
        }
        
        boolean foundFirst = false;
        for (int k = 0; k < nums.length; k++) {
            if (nums[k] == n1 && !foundFirst) {
                arr[0] = k;
                foundFirst = true;  
            } else if (nums[k] == n2) {
                arr[1] = k;
            }
        }

        return arr;
    }
    
}